### ISSUE TITLE
Programmatic control of text undo/redo

### ISSUE THREAD

**Issue by justinmc** [flutter/flutter#98280]
Recently, undo/redo (https://github.com/flutter/flutter/pull/96968) was added to Flutter's text editing.  It is contained within EditableText privately, but some developers may want to be able to control its behavior programmatically.

Currently, the only way its behavior can be affected by the developer is by using Shortcuts and Intents to affect its use of UndoTextIntent and RedoTextIntent.

CC @YeungKC

**Comment by YeungKC** [comment]
We tend pass warps _UndoStack through to EditableText from TextField provide more functionality.

Undo history be clears when submit on different platform(like macOS, iOS, web), but they not provide ignore or override that.

cc @fbcouch 
I don't know if such a plan would affect or help you.

**Comment by justinmc** [comment]
@Renzo-Olivares This reminds me of your work to allow programmatically pushing state to the undo stack.

**Comment by Renzo-Olivares** [comment]
@justinmc Yeah the work I'm doing sounds like it would also cover this issue, assigning myself.
